### PR TITLE
Add GraalVM images

### LINKED ISSUE #242: graal based image
Just putting it out there, is there any plans to support graal based image?

### PR DESCRIPTION
**Pull request by alvarosanchez**
This PR adds images that use GraalVM Community and Oracle GraalVM for JDK 17 and 21.

Fixes #242

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -200,58 +200,62 @@ This is the defacto image. If you are unsure about what your needs are, you prob
...
+| libericaopenjdk-17-alpine   |     | ✔️   | ✔️  | ✔️   | ✔️    | ✔️   | ✔️     |     |     |
+| libericaopenjdk-17-debian   |     | ✔️   | ✔️  | ✔️   | ✔️    | ✔️   |        |     |     |
+| microsoft-openjdk-11-ubuntu | ✔️  | ✔️   | ✔️  | ✔️   | ✔️    | ✔️   | ✔️     | ✔️  |     |
+| microsoft-openjdk-17-ubuntu | ✔️  | ✔️   | ✔️  | ✔️   | ✔️    | ✔️   | ✔️     | ✔️  |     |
+| microsoft-openjdk-21-ubuntu | ✔️  | ✔️   | ✔️  | ✔️   | ✔️    | ✔️   | ✔️     | ✔️  |     |
+| sapmachine-11               | ✔️  | ✔️   | ✔️  | ✔️   | ✔️    | ✔️   | ✔️     |     |     |
+| sapmachine-17               | ✔️  | ✔️   | ✔️  | ✔️   | ✔️    | ✔️   | ✔️     |     |     |
+| sapmachine-21               | ✔️  | ✔️   | ✔️  | ✔️   | ✔️    | ✔️   | ✔️     |     |     |
+| oracle-graalvm-17           |     | ✔️   | ✔️  | ✔️   | ✔️    | ✔️   | ️      | ✔️  |     |
+| oracle-graalvm-21           |     | ✔️   | ✔️  | ✔️   | ✔️    | ✔️   | ️      | ✔️  |     |
```
Comment (carlossg) [inline README.md:258]: formatting is messed up

**Review by carlossg — COMMENTED** [review]
the tests for gpg and other tools are in https://github.com/carlossg/docker-maven/blob/main/tests/tests.bats#L146

File: oracle-graalvm-17/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,19 @@
+FROM container-registry.oracle.com/graalvm/native-image:17
+
+RUN microdnf --refresh -y install findutils which
```
Comment (carlossg) [inline oracle-graalvm-17/Dockerfile:3]: in other images these are used for backwards compatibility, you may want to remove them if they are not needed and adjust the tests and readme table

Comment (alvarosanchez) [reply oracle-graalvm-17/Dockerfile:3]: Those packages are certainly not installed by default. Do you mean that they are optional?

Comment (carlossg) [reply oracle-graalvm-17/Dockerfile:3]: yes, I don't think maven needs them anymore

Comment (alvarosanchez) [reply oracle-graalvm-17/Dockerfile:3]: But `find` is used by `mvn-entrypoint.sh`

**Pull request merged**